app/src/main/java/com/example/drawingApp/MainActivity.kt

```diff
@@ -117,7 +137,27 @@ class MainActivity : AppCompatActivity() {
         );
     }
 
-    companion object {
-        private val MAX_CHAR_LEN = 16
+    fun hintText(layers: List<String>, refString: String): String {
+        if (refString != "") return refString
+        var layerCounter = 1
+        while (layers.contains(
+                resources.getString(
+                    R.string.new_layer_hint,
+                    layerCounter
+                )
+            )
+        ) {
+            layerCounter += 1
+        }
+        return resources.getString(R.string.new_layer_hint, layerCounter)
     }
+
+//    override fun onDestroy() {    //NEED TO IMPLEMENT THIS TO AVOID WINDOW LEAK
```

Comment:
Either finish the implementation or let's remove this for now

Reply:
removed for now